vars/sconsBuild.groovy

```diff
@@ -2,49 +2,201 @@
+    }
+    if (!config['no_install']) {
+        scons_args += ' install'
+    }
+    if (config['USE_INSTALLED']) {
+        scons_args += " USE_INSTALLED=${config['USE_INSTALLED']}"
+    } else {
+        scons_args += ' USE_INSTALLED=all'
+    }
+    if (config['SRC_PREFIX']) {
+        scons_args += " SRC_PREFIX=${config['SRC_PREFIX']}"
+    } else if (config['target']) {
+        scons_args += " SRC_PREFIX=\${WORKSPACE}}"
+    }
+    if (config['REQUIRES']) {
+        scons_args += " REQUIRES=${config['REQUIRES']}"
+    } else if (config['target']) {
+        scons_args += " REQUIRES=${config['target']}"
+    }
+    //scons -c is not perfect so get out the big hammer
+    clean_files = "_build.external* install build"
+    if (config['clean']) {
+        clean_files = config['clean']
+    }
+    clean_cmd = "scons -c ${sconstruct}\n"
+    if (clean_files) {
+        clean_cmd += "rm -rf ${clean_files}\n"
+        if (config['target']) {
+            clean_cmd += "rm -rf ${config['target']}/${clean_files}\n"
+        }
+    }
+
+    prebuild = ''
+    if (config['prebuild']) {
+      prebuild = config['prebuild'] + '\n'
+    }
+    prefix_1 = ''
+
+    if (config['TARGET_PREFIX']) {
+      scons_args += " TARGET_PREFIX=${config['TARGET_PREFIX']}"
+      if (config['target_work']) {
+        target_dirs = ''
+        if (config['target_dirs']) {
+          target_dirs = config['target_dirs']
+        } else if (config['target']) {
+          target_dirs = config['target']
+        }
+        target_work = config['target_work']
+        prefix_1 =
+          """for new_dir in ${target_dirs}; do
+               mkdir -p "\${WORKSPACE}/${target_work}/\${new_dir}"
+               rm -f "${config['TARGET_PREFIX']}/\${new_dir}"
+               ln -s "\${WORKSPACE}/${target_work}/\${new_dir}" \
+                 "${config['TARGET_PREFIX']}/\${new_dir}"
+             done
+             """
+      }
+    }
+    if (config['scons_args']) {
+        scons_args += ' ' + config['scons_args']
+    }
 
     script = '''if git show -s --format=%B | grep "^Skip-build: true"; then
                     exit 0
                 fi
-                scons -c
-                # scons -c is not perfect so get out the big hammer
-                rm -rf _build.external install build
-                SCONS_ARGS="--update-prereq=all --build-deps=yes USE_INSTALLED=all install"
-                # the config cache is unreliable so always force a reconfig
-                # with "--config=force"
-                if ! scons --config=force $SCONS_ARGS; then
-                    rc=\${PIPESTATUS[0]}
-                    echo "$SCONS_ARGS failed"
-                    cat config.log || true
-                    exit \$rc
-                fi'''
-    rc = sh(script: script, returnStatus: true)
-
+                '''
+    script += clean_cmd
+    script += 'SCONS_ARGS="' + scons_args +'"\n'
+    script += '''# the config cache is unreliable so always force a reconfig
```

Comment:
Is there any particular reason to keep adding to `script` instead of just continuting the multi-line string (albeit with """ instead of ''' so that you get variable interpolation)?

Reply:
Was not sure about getting string interpolation exactly right with quoting.

Reply:
Lines 152-158 right above: https://github.com/daos-stack/pipeline-lib/blob/fcc29777463e0fb09ac60a45caca01b04885209d/vars/sconsBuild.groovy#L152-L158 provides a pretty good example I think.

If you decide you want to change it.

Reply:
For now I would like to leave it the same.